Implement ProductsController#show and add request specs for Shopping Lists and Shopping List Items

robertogsf opened the pull request.

robertogsf committed:
Implement ProductsController#show and add request specs for Shopping Lists and Shopping List Items

luilver commented on .rubocop.yml:
```diff
@@ -1,3 +1,20 @@
+    - 'db/schema.rb'
+    - 'db/migrate/**/*'
+    - 'vendor/**/*'
+    - 'node_modules/**/*'
```
I would remove this exclusion.

robertogsf replied:
ready

luilver commented on CHANGELOG.md:
```diff
@@ -0,0 +1,43 @@
+
+All notable changes to this project will be documented in this file.
+
+The format is based on Keep a Changelog and this project adheres to Semantic Versioning.
```
I would include SemVer version (1.1.0)

luilver replied:
I mean: 
```
The format is based on Keep a Changelog and this project adheres to Semantic Versioning version 1.1.0.
```

robertogsf replied:
then I delete the bug resolution logs

luilver commented on CHANGELOG.md:
```diff
@@ -0,0 +1,43 @@
+### Added
+- Request specs para Shopping Lists (index, create, show, update, delete).
+- Request spec para Shopping List Items destroy.
+- Request specs extra para Products: show y filtro por store_id.
```
I would remove Spanglish.

robertogsf replied:
ready

luilver commented on CHANGELOG.md:
```diff
@@ -0,0 +1,43 @@
+- Request specs extra para Products: show y filtro por store_id.
+
+### Changed
+- Implementado `ProductsController#show` para GET /api/v1/products/:id.
```
I would remove Spanglish.

robertogsf replied:
ready

luilver commented on app/controllers/api/v1/products_controller.rb:
```diff
@@ -16,6 +16,16 @@ def index
 
+  # GET /api/v1/products/:id
+  def show
+    product = Product.find_by(id: params[:id])
```
I would adhere to strong parameters here.

robertogsf replied:
ready

luilver commented on app/controllers/api/v1/shopping_lists_controller.rb:
```diff
@@ -2,7 +2,7 @@ module Api
     class ShoppingListsController < ApplicationController
       before_action :authorize_user
-      before_action :set_shopping_list, only: [:show, :update, :destroy]
+  before_action :set_shopping_list, only: [ :show, :update, :destroy ]
```
I would restore previous indentation level here.

robertogsf replied:
ready

luilver commented on app/controllers/api/v1/shopping_lists_controller.rb:
```diff
@@ -45,7 +45,7 @@ def destroy
       def set_shopping_list
         @shopping_list = current_user.shopping_lists.find_by(id: params[:id])
-        return render json: { error: 'Not Found' }, status: :not_found unless @shopping_list
+  render json: { error: "Not Found" }, status: :not_found unless @shopping_list
```
I would restore previous indentation level here.

robertogsf replied:
ready

luilver commented on spec/integration/products_extra_spec.rb:
```diff
@@ -0,0 +1,51 @@
+require 'swagger_helper'
```
I would use the existing products_spec.rb file.

robertogsf replied:
unified the files and removed the extra file

luilver commented on spec/integration/products_spec.rb:
```diff
@@ -7,7 +7,7 @@
       security [ bearer_auth: [] ]
 
-      response '200', 'products found' do
+  response '200', 'products found' do
```
I would restore previous indentation level here.

robertogsf replied:
ready

luilver commented on spec/integration/shopping_list_items_destroy_spec.rb:
```diff
@@ -0,0 +1,31 @@
+require 'swagger_helper'
```
I would use the existing shopping_list_items_spec.rb file unless you have a strong opinion on different files for different endpoints. Open to a discussion.

robertogsf replied:
You're right, I keep the structure and unify the files.

luilver commented on spec/integration/shopping_list_items_spec.rb:
```diff
@@ -20,7 +20,7 @@
 
       response '201', 'item added' do
-        let(:store) { Store.create!(name: "Test Store") }
+  let(:store) { Store.create!(name: "Test Store") }
```
I would restore previous indentation level here.

robertogsf replied:
ready

luilver commented on spec/integration/shopping_lists_spec.rb:
```diff
@@ -0,0 +1,127 @@
+          token = JWT.encode({ user_id: user.id }, secret, 'HS256')
+          "Bearer #{token}"
+        end
+
```
I would remove this empty line.

robertogsf replied:
ready

luilver commented on spec/integration/products_extra_spec.rb:
```diff
@@ -0,0 +1,51 @@
+        let!(:sp) { StoreProduct.create!(store: store, product: product, aisle: '1', location: { shelf: 1 }, price: 1.23) }
+        let(:store_id) { store.id }
+
+        run_test!
```
I would include, and won't limit to, a couple expectations:

1. HTTP response status
2. response body should be a list of products

luilver commented on spec/integration/products_extra_spec.rb:
```diff
@@ -0,0 +1,51 @@
+        let!(:product) { Product.create!(name: 'Bananas', category: 'Produce', brand: 'Dole', image_url: 'x') }
+        let(:id) { product.id }
+
+        run_test!
```
I would include, and won't limit to, a couple expectations:

1. HTTP response status
2. response body should be a product

luilver commented on spec/integration/shopping_list_items_destroy_spec.rb:
```diff
@@ -0,0 +1,31 @@
+        let(:shopping_list_id) { list.id }
+        let(:id) { item.id }
+
+        run_test!
```
I would include, and won't limit to, a couple expectations:

1. HTTP response status
2. response body should be a shopping list item

luilver commented on spec/integration/shopping_lists_spec.rb:
```diff
@@ -0,0 +1,127 @@
+        let!(:store) { Store.create!(name: 'Test Store') }
+        let!(:list1) { ShoppingList.create!(user: user, store: store, name: 'Weekly') }
+
+        run_test!
```
I would include, and won't limit to, a couple expectations:

1. HTTP response status
2. response body should be a list of shopping lists

luilver commented on spec/integration/shopping_lists_spec.rb:
```diff
@@ -0,0 +1,127 @@
+      produces 'application/json'
+      security [ bearer_auth: [] ]
+
+      response '200', 'lists found' do
```
I would use 'Shopping lists found' instead.

luilver commented on spec/integration/shopping_lists_spec.rb:
```diff
@@ -0,0 +1,127 @@
+        }
+      }
+
+      response '201', 'list created' do
```
I would use 'Shopping lists created' instead.

luilver commented on spec/integration/shopping_lists_spec.rb:
```diff
@@ -0,0 +1,127 @@
+      produces 'application/json'
+      security [ bearer_auth: [] ]
+
+      response '200', 'list shown' do
```
I would use 'Shopping lists shown' instead.

luilver commented on spec/integration/shopping_lists_spec.rb:
```diff
@@ -0,0 +1,127 @@
+        }
+      }
+
+      response '200', 'list updated' do
```
I would use 'Shopping lists updated' instead.

luilver commented on spec/integration/shopping_lists_spec.rb:
```diff
@@ -0,0 +1,127 @@
+      produces 'application/json'
+      security [ bearer_auth: [] ]
+
+      response '204', 'list deleted' do
```
I would use 'Shopping lists deleted' instead.

luilver commented on spec/integration/shopping_lists_spec.rb:
```diff
@@ -0,0 +1,127 @@
+        let!(:store) { Store.create!(name: 'Store A') }
+        let(:shopping_list) { { name: 'Party', store_id: store.id } }
+
+        run_test!
```
I would include, and won't limit to, a couple expectations:

1. HTTP response status
2. response body should be a shopping list

luilver commented on spec/integration/shopping_lists_spec.rb:
```diff
@@ -0,0 +1,127 @@
+        let!(:list) { ShoppingList.create!(user: user, store: store, name: 'Groceries') }
+        let(:id) { list.id }
+
+        run_test!
```
I would include, and won't limit to, a couple expectations:

1. HTTP response status
2. response body should be a shopping list

luilver commented on spec/integration/shopping_lists_spec.rb:
```diff
@@ -0,0 +1,127 @@
+        let(:id) { list.id }
+        let(:shopping_list) { { name: 'Renamed' } }
+
+        run_test!
```
I would include, and won't limit to, a couple expectations:

1. HTTP response status
2. response body should be a shopping list

luilver commented on spec/integration/shopping_lists_spec.rb:
```diff
@@ -0,0 +1,127 @@
+        let!(:list) { ShoppingList.create!(user: user, store: store, name: 'Temp') }
+        let(:id) { list.id }
+
+        run_test!
```
I would include, and won't limit to, a couple expectations:

1. HTTP response status
2. response body should be nil

robertogsf committed:
Refactor ProductsController and ShoppingListsController for improved parameter handling and response messages; update request specs for Shopping Lists to validate response structure.

robertogsf committed:
wip

luilver commented on .rubocop.yml:
```diff
@@ -7,10 +7,6 @@ AllCops:
     - 'bin/**/*'
-    - 'db/schema.rb'
-    - 'db/migrate/**/*'
-    - 'vendor/**/*'
```
I wouldn't remove these three.

luilver commented on CHANGELOG.md:
```diff
@@ -14,15 +14,15 @@ The format is based on Keep a Changelog and this project adheres to Semantic Ver
 <!-- When releasing, move items above into a new section, e.g. `## [0.1.0] - YYYY-MM-DD` -->
 
-## [0.1.1] - 2025-08-25
+## [1.1.0] - 2025-08-25
```
Any reason behind this change?

robertogsf replied:
https://github.com/luilver/aisle_pilot_api/pull/9#discussion_r2300083792

luilver replied:
That (1.1.0) is the version of SameVer we're using, not the version of this project.

luilver commented on app/controllers/api/v1/shopping_lists_controller.rb:
```diff
@@ -37,8 +37,8 @@ def update
-        @shopping_list.destroy
-        head :no_content
+      @shopping_list.destroy
+      render json: { message: "Shopping lists deleted" }, status: :ok
```
I would respect indentation level.

luilver replied:
This is still a thing

robertogsf committed:
merge

robertogsf committed:
fix CRLF

robertogsf committed:
fix CRLF

robertogsf committed:
style: fix various formatting issues and rubocop config

- Remove trailing whitespace in multiple files
- Fix array bracket spacing in controller and spec files
- Add missing newline at end of file
- Reorganize rubocop.yml configuration
- Update CHANGELOG.md with formatting fixes

luilver commented on .rubocop.yml:
```diff
@@ -1,5 +1,17 @@
+  NewCops: enable
+  Exclude:
+    - 'bin/**/*'
+    - 'tmp/**/*'
```
I would remove only the line with node_modules and restore:
```
    - 'db/schema.rb'
    - 'db/migrate/**/*'
    - 'vendor/**/*'
```

luilver commented on app/controllers/api/v1/products_controller.rb:
```diff
@@ -16,6 +16,20 @@ def index
+
+  # GET /api/v1/products/:id
+  def show
+  product = Product.find_by(id: show_params[:id])
```
I would respect indentation level.

robertogsf committed:
refactor(tests): consolidate test files and improve test descriptions

- Merge products_extra_spec into products_spec
- Merge shopping_list_items_destroy_spec into shopping_list_items_spec
- Update test descriptions to be more consistent and descriptive
- Fix formatting and security syntax in test files

robertogsf committed:
fix changelog

robertogsf committed:
style: fix indentation in controllers and update rubocop exclusions

Adjust indentation in products and shopping lists controllers to maintain consistent style. Also exclude additional files (schema.rb, migrations, vendor) from rubocop checks to avoid unnecessary linting on generated files.

luilver commented on CHANGELOG.md:
```diff
@@ -0,0 +1,48 @@
+
+The format is based on Keep a Changelog and this project adheres to Semantic Versioning.
+
+## [1.1.0] - 2025-08-25
```
version is 0.1.2

luilver commented on app/controllers/api/v1/products_controller.rb:
```diff
@@ -16,6 +16,20 @@ def index
+
+  # GET /api/v1/products/:id
+  def show
+      product = Product.find_by(id: show_params[:id])
```
Indentation level too deep.

luilver commented on app/controllers/api/v1/shopping_lists_controller.rb:
```diff
@@ -38,7 +38,38 @@ def update
       def destroy
         @shopping_list.destroy
-        head :no_content
+          render json: { message: "Shopping lists deleted" }, status: :ok
```
Indentation level too deep.

luilver reviewed (commented):
Everything else looks good.

robertogsf committed:
style: fix indentation in products and shopping lists controllers

robertogsf committed:
fix sem version

robertogsf committed:
feat: enhance JWT secret handling and update API documentation for stores and shopping lists
